tools/CMakeLists.txt

```diff
@@ -0,0 +1,46 @@
+# ~~~
+# Create a header-only library from an SVD register definition file.
+#
+# add_groov_reg_lib(
+#     groov_reg_test
+#     INPUT_FILE "${CMAKE_SOURCE_DIR}/regs/rp2040.svd"
+#     OUTPUT_DIR "${CMAKE_BINARY_DIR}/regs"
+#     PERIPHERALS adc rtc
+# )
+# ~~~
+
+function(add_groov_reg_lib target_name)
+    message(ERROR "Python3 not found, `add_groov_reg_lib` is not available.")
+endfunction()
+
+if(NOT Python3_FOUND)
+    return()
+endif()
+
+function(add_groov_reg_lib target_name)
+    set(oneValueArgs INPUT_FILE OUTPUT_DIR)
+    set(multiValueArgs PERIPHERALS)
+    cmake_parse_arguments(ARG "" "${oneValueArgs}" "${multiValueArgs}" ${ARGN})
+
+    string(JOIN " " peripherals_string "${ARG_PERIPHERALS}")
+    set(python_script "${CMAKE_CURRENT_FUNCTION_LIST_DIR}/regs2groov.py")
+
+    list(TRANSFORM ARG_PERIPHERALS APPEND ".hpp" OUTPUT_VARIABLE
+                                                 register_headers)
+
+    add_custom_command(
+        OUTPUT ${register_headers}
+        COMMAND
+            ${Python3_EXECUTABLE} ${python_script} --input ${ARG_INPUT_FILE}
+            --output ${ARG_OUTPUT_DIR} --peripherals ${peripherals_string}
```

Comment:
**suggestion**: Just noticed this: instead of making the string on line 25, here you can do `--peripherals ${ARG_PERIPHERALS}` and use `COMMAND_EXPAND_LISTS`.

Reply:
i like it, just made the change